### PR TITLE
fix: to_rpc_contract_class ✨

### PR DESCRIPTION
**Pull request by EvolveArt**
## What is the current behavior?

Resolves: #363 

## What is the new behavior?

Fixes `to_rpc_contract_class`, converts to a legacy contract class.

## Does this introduce a breaking change?

No

### CODE REVIEW

**Comment by EvolveArt** [comment]
We need to wait for this to be fixed https://github.com/lambdaclass/cairo-rs/issues/1183

File: crates/client/rpc-core/src/utils.rs [hunk 1]
```diff
@@ -161,33 +177,51 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
...
+        .collect::<Result<Vec<LegacyContractEntryPoint>, FromByteArrayError>>()?;
+
+    let l1_handler = entries
+        .get(&EntryPointTypeWrapper::L1Handler)
+        .unwrap()
+        .iter()
+        .map(|e| (e.clone()).try_into())
+        .collect::<Result<Vec<LegacyContractEntryPoint>, FromByteArrayError>>()?;
+
+    Ok(LegacyEntryPointsByType { constructor, external, l1_handler })
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:217]: remove unwraps

File: crates/client/rpc-core/src/utils.rs [hunk 2]
```diff
@@ -161,33 +177,51 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
...
-            .unwrap(),
-    );
+
+    entry_points_by_type.insert(EntryPointTypeWrapper::Constructor, _get_entrypoint_value(entries.constructor));
+    entry_points_by_type.insert(EntryPointTypeWrapper::External, _get_entrypoint_value(entries.external));
+    entry_points_by_type.insert(EntryPointTypeWrapper::L1Handler, _get_entrypoint_value(entries.l1_handler));
     entry_points_by_type
 }
 
+fn _to_legacy_entry_points_by_type(
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:193]:
```suggestion
fn to_legacy_entry_points_by_type(
```

File: crates/client/rpc-core/src/utils.rs [hunk 3]
```diff
@@ -161,33 +177,51 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
...
+        .unwrap()
+        .iter()
+        .map(|e| (e.clone()).try_into())
+        .collect::<Result<Vec<LegacyContractEntryPoint>, FromByteArrayError>>()?;
+
+    Ok(LegacyEntryPointsByType { constructor, external, l1_handler })
+}
+
+/// Returns a bounded vector of `EntryPointWrapper` from a vector of LegacyContractEntryPoint
+fn _get_entrypoint_value(entries: Vec<LegacyContractEntryPoint>) -> BoundedVec<EntryPointWrapper, MaxEntryPoints> {
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:221]:
```suggestion
fn get_entrypoint_value(entries: Vec<LegacyContractEntryPoint>) -> BoundedVec<EntryPointWrapper, MaxEntryPoints> {
```

File: crates/client/rpc-core/src/utils.rs [hunk 4]
```diff
@@ -161,33 +177,51 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
     }
 }
 
-/// Returns a hash map of entry point types to entrypoint from deprecated entry point by type
+/// Returns a btree map of entry point types to entrypoint from deprecated entry point by type
 fn _to_btree_map_entrypoints(
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:181]:
```suggestion
fn to_btree_map_entrypoints(
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: crates/client/rpc-core/src/utils.rs [hunk 5]
```diff
@@ -161,33 +177,50 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
     }
 }
 
-/// Returns a hash map of entry point types to entrypoint from deprecated entry point by type
-fn _to_btree_map_entrypoints(
+/// Returns a btree map of entry point types to entrypoint from deprecated entry point by type
+fn to_btree_map_entrypoints(
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:181]: impl from trait ?

File: crates/client/rpc-core/src/utils.rs [hunk 6]
```diff
@@ -161,33 +177,50 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
...
-            .unwrap(),
-    );
+
+    entry_points_by_type.insert(EntryPointTypeWrapper::Constructor, get_entrypoint_value(entries.constructor));
+    entry_points_by_type.insert(EntryPointTypeWrapper::External, get_entrypoint_value(entries.external));
+    entry_points_by_type.insert(EntryPointTypeWrapper::L1Handler, get_entrypoint_value(entries.l1_handler));
     entry_points_by_type
 }
 
+fn to_legacy_entry_points_by_type(
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:193]: impl from trait ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: crates/client/rpc-core/src/utils.rs [hunk 7]
```diff
@@ -83,6 +97,7 @@ pub fn to_invoke_tx(tx: BroadcastedInvokeTransaction) -> Result<InvokeTransactio
     }
 }
 
+/// Converts a broadcasted deploy account transaction to a deploy account transaction
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:100]: from trait ?

File: crates/client/rpc-core/src/utils.rs [hunk 8]
```diff
@@ -118,6 +133,7 @@ pub fn to_deploy_account_tx(tx: BroadcastedDeployAccountTransaction) -> Result<D
     })
 }
 
+/// Converts a broadcasted declare transaction to a declare transaction
```
Comment (LucasLvy) [inline crates/client/rpc-core/src/utils.rs:136]: from trait ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**